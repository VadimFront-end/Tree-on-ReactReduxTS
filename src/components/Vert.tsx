import {TActionType, IVert} from "../interfaces";
import React from "react";
import {bindActionCreators, Dispatch} from "redux";
import {add_right_vert, add_left_vert, delete_vert} from "../store/actions";
import {connect} from "react-redux";

export type props = {
    vert: IVert,
    add_right_vert(vert: number): TActionType,
    add_left_vert(vert: number): TActionType,
    delete_vert(vert: number): TActionType
}

export const Vert = ({vert, add_right_vert, add_left_vert, delete_vert}: props) => {
    return (
        <div style={{width: vert.lvl === 0 ? '100%' : '50%'}}>
            {
                vert.lvl !== 0 ? <div className="bw-lvl" style={{justifyContent: vert.parent!.left === vert ? 'flex-end': 'flex-start'}}>
                    <div className="arc" style={{transform: vert.parent!.left === vert ? `rotate(-${7*vert.lvl}deg)`: `rotate(${7*vert.lvl}deg)`, width: '50%'}}></div>
                </div>: ''
            }
            <div className="vert-wrapper">
                <div style={{display: "flex", flexDirection: "column", alignItems: "center"}}>
                    <div className="vert">{vert.id}
                        <div style={{cursor: "pointer", display: vert.id === 0 ? 'none' : ''}}
                             onClick={() => delete_vert(vert.id)}>&#9746;</div>
                    </div>
                    <div>
                        <button onClick={() => add_left_vert(vert.id)}
                                style={{cursor: "pointer", visibility: vert.left ? 'hidden' : 'visible'}}>L
                        </button>
                        <button onClick={() => add_right_vert(vert.id)}
                                style={{cursor: "pointer", visibility: vert.right ? 'hidden' : 'visible'}}>R
                        </button>
                    </div>
                </div>
            </div>
            <div style={{display: "flex", justifyContent: "space-evenly"}}>
                {vert.left ? <Vert vert={vert.left} add_right_vert={add_right_vert} add_left_vert={add_left_vert}
                                   delete_vert={delete_vert}/> : <div style={{width: '50%'}}></div>}
                {vert.right ? <Vert vert={vert.right} add_left_vert={add_left_vert} add_right_vert={add_right_vert}
                                    delete_vert={delete_vert}/> : <div style={{width: '50%'}}></div>}
            </div>
        </div>
    );
}

const mapDispatchToProps = (dispatch: Dispatch) => {
    return {
        add_right_vert: bindActionCreators(add_right_vert, dispatch),
        add_left_vert: bindActionCreators(add_left_vert, dispatch),
        delete_vert: bindActionCreators(delete_vert, dispatch)
    }
}

export default connect(null, mapDispatchToProps)(Vert);
