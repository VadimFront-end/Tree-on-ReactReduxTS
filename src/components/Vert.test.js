import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {Vert} from "./Vert";
Enzyme.configure({ adapter: new Adapter() });

describe('Testing component `Vert`', () => {
    const testRightVert = {
        id: 1,
        left: null,
        right: null,
        lvl: 1,
        parent: null
    }
    const testVert = {
        id: 0,
        left: null,
        right: testRightVert,
        lvl: 0,
        parent: null
    }
    testRightVert.parent = testVert;
    it('Can i test component with Snapshot', () => {
        const component = renderer.create(
            <Vert vert={testVert}/>
        );
        expect(component).toMatchSnapshot();
    })

    // it('Can i test component with enzyme', () => {
    //     const component = shallow(<Vert vert={testVert} />);
    //     console.log(component.debug())
    //     expect(component).toBe(1);
    // });
    //доделать
})
