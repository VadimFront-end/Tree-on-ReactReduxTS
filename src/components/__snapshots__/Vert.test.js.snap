// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component \`Vert\` Can i test component with Snapshot 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  
  <div
    className="vert-wrapper"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <div
        className="vert"
      >
        0
        <div
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "display": "none",
            }
          }
        >
          ☒
        </div>
      </div>
      <div>
        <button
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "visibility": "visible",
            }
          }
        >
          L
        </button>
        <button
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "visibility": "hidden",
            }
          }
        >
          R
        </button>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-evenly",
      }
    }
  >
    <div
      style={
        Object {
          "width": "50%",
        }
      }
    />
    <div
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <div
        className="bw-lvl"
        style={
          Object {
            "justifyContent": "flex-start",
          }
        }
      >
        <div
          className="arc"
          style={
            Object {
              "transform": "rotate(7deg)",
              "width": "50%",
            }
          }
        />
      </div>
      <div
        className="vert-wrapper"
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <div
            className="vert"
          >
            1
            <div
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "display": "",
                }
              }
            >
              ☒
            </div>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "visibility": "visible",
                }
              }
            >
              L
            </button>
            <button
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "visibility": "visible",
                }
              }
            >
              R
            </button>
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-evenly",
          }
        }
      >
        <div
          style={
            Object {
              "width": "50%",
            }
          }
        />
        <div
          style={
            Object {
              "width": "50%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
